cwms-data-api/src/main/java/cwms/cda/data/dao/TimeSeriesVerticalDatumConverter.java

```diff
@@ -0,0 +1,128 @@
+package cwms.cda.data.dao;
+
+import cwms.cda.data.dto.TimeSeries;
+import cwms.cda.data.dto.VerticalDatumInfo;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public final class TimeSeriesVerticalDatumConverter {
+
+
+    private TimeSeriesVerticalDatumConverter() {
+        throw new AssertionError("Utility class, don't instantiate");
+    }
+
+    public static TimeSeries convertToVerticalDatum(TimeSeries originalTimeSeries, VerticalDatum convertTo) {
+        if(originalTimeSeries.getVerticalDatumInfo() == null || Objects.equals(convertTo, getVerticalDatum(originalTimeSeries))) {
+            return originalTimeSeries; //no conversion needed
+        }
+        TimeSeries retVal = originalTimeSeries;
+        VerticalDatumInfo vdi = originalTimeSeries.getVerticalDatumInfo();
+        VerticalDatumInfo.Offset offset = getOffsetForDatum(vdi, convertTo);
+        if(offset != null)
+        {
+            List<TimeSeries.Record> newValues = convertElevationValues(offset, originalTimeSeries);
+            VerticalDatumInfo newVerticalDatumInfo = convertVerticalDatumInfo(vdi, convertTo, offset);
+            retVal = new TimeSeries(originalTimeSeries.getPage(),
+                    originalTimeSeries.getPageSize(),
+                    originalTimeSeries.getTotal(),
+                    originalTimeSeries.getName(),
+                    originalTimeSeries.getOfficeId(),
+                    originalTimeSeries.getBegin(),
+                    originalTimeSeries.getEnd(),
+                    originalTimeSeries.getUnits(),
+                    originalTimeSeries.getInterval(),
+                    newVerticalDatumInfo,
+                    originalTimeSeries.getIntervalOffset(),
+                    originalTimeSeries.getTimeZone(),
+                    originalTimeSeries.getVersionDate(),
+                    originalTimeSeries.getDateVersionType())
+                    .withValues(newValues);
+        }
+        return retVal;
+    }
+
+    @NotNull
+    private static List<TimeSeries.Record> convertElevationValues(VerticalDatumInfo.Offset offset, TimeSeries retVal) {
```

Comment:
why is elevation in the name?  it doesn't check if the TimeSeries is elev.   Also the TimeSeries parameter being called retVal isn't a great name